crates/autopilot/src/infra/persistence/dto/auction.rs

```diff
@@ -72,22 +49,44 @@ pub struct Auction {
 
 pub type AuctionId = i64;
 
-impl TryFrom<AuctionWithId> for domain::AuctionWithId {
+impl TryFrom<Auction> for domain::Auction {
     type Error = anyhow::Error;
 
-    fn try_from(dto: AuctionWithId) -> anyhow::Result<Self> {
-        Ok(domain::AuctionWithId {
+    fn try_from(dto: Auction) -> anyhow::Result<Self> {
+        Ok(domain::Auction {
             id: dto.id,
-            auction: try_to_domain(dto.auction)?,
+            block: dto.auction.block,
+            latest_settlement_block: dto.auction.latest_settlement_block,
+            orders: dto
+                .auction
+                .orders
+                .into_iter()
+                .map(super::order::to_domain)
+                .collect(),
+            prices: dto
+                .auction
+                .prices
+                .into_iter()
+                .map(|(key, value)| {
+                    Price::new(value.into()).map(|price| (eth::TokenAddress(key), price))
+                })
+                .collect::<Result<_, _>>()?,
+            surplus_capturing_jit_order_owners: dto
+                .auction
+                .surplus_capturing_jit_order_owners
+                .into_iter()
+                .map(Into::into)
+                .collect(),
         })
     }
 }
 
 #[serde_as]
 #[derive(Clone, Debug, Deserialize)]
+#[serde(rename = "AuctionWithId")]
```

Comment:
Same here: not sure if this is actually needed since this name should never show up anywhere in JSON AFAIK. 🤔

Reply:
removed